### DISCUSSION TITLE
How to access query params passed to api action (dynamic routes)

### DISCUSSION THREAD

**Discussion by borsemayur2** [alephjs/aleph.js#140] [Q&A] [answered]
I'm writing Todos APIs. I need to pass `id` of the todo record for `retrieve`, `update` and `delete` action. 

How Should I pass the `id` ?
`/todos/retrieve/1 ` or `/todos/retrieve?id=1`

How can I access the passed `id`?
request.params contains `{action: 'retrieve'}` only

**Comment by shadowtime2000** [comment, marked as answer]
Please use discussions for questions. You can use the `useRouter` hook to collect information from queries and other stuff.

**Reply by borsemayur2** [reply]
Okay. Thanks

**Comment by ije** [comment]
`useRouter` in the `/todos/retrieve/[id].tsx`
